polkadot/runtime/rococo/src/impls.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::xcm_config;
+use frame_support::pallet_prelude::DispatchResult;
+use frame_system::RawOrigin;
+use pallet_identity::OnReapIdentity;
+use primitives::Balance;
+use rococo_runtime_constants::currency::*;
+use sp_std::{marker::PhantomData, prelude::*};
+use xcm::{latest::prelude::*, /* v2::OriginKind, */ VersionedMultiLocation, VersionedXcm};
+use xcm_executor::traits::TransactAsset;
+
+/// Type that implements `OnReapIdentity` that will send the deposit needed to store the same
+/// information on a parachain, sends the deposit there, and then updates it.
+pub struct ToParachainIdentityReaper<T>(PhantomData<T>);
+impl<AccountId, T> OnReapIdentity<AccountId> for ToParachainIdentityReaper<T>
+where
+	T: frame_system::Config + pallet_xcm::Config,
+	AccountId: Into<[u8; 32]> + Clone,
+{
+	fn on_reap_identity(who: &AccountId, fields: u32, subs: u32) -> DispatchResult {
+		// calculate deposit needed on parachains
+		// assume `relay_deposit / 100 + ED + buffer`
+		// buffer to cover tx fees on para, maybe ED
+		let para_basic_deposit = 1000 * CENTS / 100;
+		let para_field_deposit = 250 * CENTS / 100;
+		let para_sub_account_deposit = 200 * CENTS / 100;
+		let para_existential_deposit = EXISTENTIAL_DEPOSIT / 10;
+		let id_deposit =
+			para_basic_deposit.saturating_add(para_field_deposit.saturating_mul(fields as Balance));
+
+		let subs_deposit = para_sub_account_deposit.saturating_mul(subs as Balance);
+
+		let total_to_send = id_deposit
+			.saturating_add(subs_deposit)
+			.saturating_add(para_existential_deposit.saturating_mul(2));
```

Comment:
... and have this computation in a separate function which uses those constants
```
impl ToParachainIdentityReaper {
fn total_amount(additional_fields: usize, subs: usize) -> Balance
}
```
to segregate that from the actual XCM stuff here.